src/stateStream.ts

```diff
@@ -0,0 +1,82 @@
+import { markName, defaultErrorSubject } from 'rxbeach/internal';
+import { ActionStream } from 'rxbeach/types/helpers';
+import { tag } from 'rxjs-spy/operators';
+import {
+  shareReplay,
+  startWith,
+  switchMap,
+  catchError,
+  skip,
+  filter,
+  tap,
+} from 'rxjs/operators';
+import { combineReducers, RegisteredReducer } from 'rxbeach/reducer';
+import { Subject, of } from 'rxjs';
+
+/**
+ * Create a state stream for a set of reducers.
+ *
+ * The state stream will always emit its last value upon subscription.
+ * On the first subscription to the state stream, it will emit `defaultState`.
+ *
+ * The state stream is not ref counted, which means that the underlying
+ * producer never resets to the initial value. Subsequent subscriptions to
+ * the stream will always get the previously emitted state.
+ *
+ * @param name A name for debugging purposes
+ * @param action$ The action stream
+ * @param defaultState The initial state of the state stream,
+ *                     which is emitted synchronously upon subscription
+ * @param reducers The reducer entries that are combined with `combineReducers`
+ * @see rxbeach.combineReducers
+ * @returns An stream that emits the reduced state
+ */
+export const stateStream = <State>(
+  name: string,
+  action$: ActionStream,
+  defaultState: State,
+  reducers: RegisteredReducer<State, any>[]
+) =>
+  action$.pipe(
+    combineReducers(defaultState, reducers),
+    startWith(defaultState),
+    shareReplay({
+      refCount: false,
+      bufferSize: 1,
+    }),
+    markName(name),
+    tag(name)
+  );
+
+export const stateStreamWithErrorHandling = <State>(
```

Comment:
This is obviously WIP, as I would like for this error handling to be a part of `stateStream`, however it turned out to be quite tricky to implement the desired behaviour (see the last test).

Reply:
Yeah, this implementation is horrendous :fearful: 

I'll give it a spin and see if I can do something better. 

I hope the plan is to merge this and the `stateStream` function?

Reply:
> I hope the plan is to merge this and the stateStream function?
Yes :) I split it out because it became so ugly 😅 Error handling and continuing, while emitting the correct things is a bit tricky with observables.